From: Martin Schwidefsky <schwidefsky@de.ibm.com>
Subject: IBM Codedrop 2006-01-09 - linux-2.6.15-s390-02-03-october2005.diff

  Description: cio: chps[] array too short.
  Symptom:     When several devices on chpid ff become available,
               "new_channel_path: could not register ff" is displayed for every
               device but the first one.
  Problem:     The chps[] array in struct channel_subsystem is one too short;
               therefore the code doesn't realize the chpid ff is already known.
  Solution:    Make chps[] big enough.
  Problem-ID:  -
  Archs:       s390-31, s390-64

Acked-by: Hannes Reinecke <hare@suse.de>

--- linux-2.5/drivers/s390/cio/css.h	10 Nov 2005 15:53:54 -0000	1.44
+++ linux-2.5/drivers/s390/cio/css.h	12 Jan 2006 14:11:23 -0000	1.44.2.1
@@ -143,7 +143,7 @@
 struct channel_subsystem {
 	u8 cssid;
 	int valid;
-	struct channel_path *chps[__MAX_CHPID];
+	struct channel_path *chps[__MAX_CHPID + 1];
 	struct device device;
 	struct pgid global_pgid;
 };

